test/System.IdentityModel.Tokens.Jwt.Tests/JwtSecurityTokenHandlerTests.cs

```diff
@@ -463,6 +464,99 @@ public void InboundOutboundClaimTypeMapping()
             RunClaimMappingVariation(jwt, handler, validationParameters, expectedClaims: expectedClaims, identityName: null);
         }
 
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+        [Theory, MemberData("ReadJwtTokenTheoryData")]
+#pragma warning restore CS3016 // Arrays as attribute arguments is not CLS-compliant
+        public void ReadJwtToken(JwtTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ReadJwtToken", theoryData);
+            var context = new CompareContext($"{this}.ReadJwtToken, {theoryData.TestId}");
+            try
+            {
+                var principal = theoryData.TokenHandler.ValidateToken(theoryData.Token, theoryData.ValidationParameters, out SecurityToken validToken);
+
+                var jwtToken1 = validToken as JwtSecurityToken;
+                var jwtToken2 = theoryData.SecurityToken as JwtSecurityToken;
+
+                string time1 = Convert.ToString(jwtToken1.Payload[JwtRegisteredClaimNames.Nbf], CultureInfo.InvariantCulture);
```

Comment:
Why do we need to convert to string to compare?
If the IdentityComparer will not compare two doubles, we need to fix that.

Reply:
Assign the value as "1471924298.67746", then compare.